data/debian/sonic-host-services-data.process-reboot-cause.timer

```diff
@@ -1,7 +1,9 @@
 [Unit]
 Description=Delays process-reboot-cause until network is stably connected
+PartOf=process-reboot-cause.service
 
 [Timer]
+OnUnitActiveSec=0 sec
```

Comment:
When the process-reboot-cause service runs and becomes active, systemd does not trigger any more timer. This is how its being used for other services as well like lldp@.timer

Here is a snapshot when the service is active:
```
root@sonic:/home/cisco# systemctl list-timers 
NEXT                        LEFT          LAST                        PASSED        UNIT                         ACTIVATES
n/a                         n/a           Mon 2023-11-27 20:38:33 UTC 11ms ago      snmp.timer                   snmp.service
Mon 2023-11-27 20:40:00 UTC 1min 26s left Mon 2023-11-27 20:30:01 UTC 8min ago      logrotate.timer              logrotate.service
Mon 2023-11-27 20:40:00 UTC 1min 26s left Mon 2023-11-27 20:30:01 UTC 8min ago      sysstat-collect.timer        sysstat-collect.service
Mon 2023-11-27 22:36:29 UTC 1h 57min left Sun 2023-11-26 22:36:29 UTC 22h ago       systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Tue 2023-11-28 00:07:00 UTC 3h 28min left Mon 2023-11-27 00:07:00 UTC 20h ago       sysstat-summary.timer        sysstat-summary.service
Tue 2023-11-28 06:24:11 UTC 9h left       Mon 2023-11-27 06:19:57 UTC 14h ago       apt-daily-upgrade.timer      apt-daily-upgrade.service
Tue 2023-11-28 13:10:39 UTC 16h left      Mon 2023-11-27 20:01:01 UTC 37min ago     apt-daily.timer              apt-daily.service
Sun 2023-12-03 03:10:27 UTC 5 days left   Sun 2023-11-26 03:10:43 UTC 1 day 17h ago e2scrub_all.timer            e2scrub_all.service
Mon 2023-12-04 00:00:00 UTC 6 days left   Mon 2023-11-27 00:00:00 UTC 20h ago       fstrim.timer                 fstrim.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     hostcfgd.timer               hostcfgd.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     lldp.timer                   lldp.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     lldp@0.timer                 lldp@0.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     lldp@1.timer                 lldp@1.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     mgmt-framework.timer         mgmt-framework.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     pmon.timer                   pmon.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     process-reboot-cause.timer   process-reboot-cause.service
n/a                         n/a           Wed 2023-11-22 22:22:47 UTC 4 days ago    rasdaemon.timer              rasdaemon.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     tacacs-config.timer          tacacs-config.service
n/a                         n/a           Mon 2023-11-27 20:00:48 UTC 37min ago     telemetry.timer              telemetry.service

19 timers listed.
Pass --all to see loaded but inactive timers, too.
root@sonic:/home/cisco# systemctl status process-reboot-cause.service 
● process-reboot-cause.service - Retrieve the reboot cause from the history files and save them to StateDB
     Loaded: loaded (/lib/systemd/system/process-reboot-cause.service; enabled; vendor preset: enabled)
     Active: active (exited) since Mon 2023-11-27 20:00:50 UTC; 37min ago
TriggeredBy: ● process-reboot-cause.timer
   Main PID: 3946159 (code=exited, status=0/SUCCESS)
   ```
   
In case database service has failed causing process-reboot-cause dependency failure, systemd uses timer to trigger every sec until the service has started. I have not seen any adverse effect of that on system cpui usage and it should not happen in general.
```
19 timers listed.                                                                                                                            
Pass --all to see loaded but inactive timers, too.                                                                                        
root@sonic:/home/cisco# systemctl list-timers                                                                                                  
NEXT                        LEFT          LAST                        PASSED        UNIT                         ACTIVATES              
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 122ms ago     hostcfgd.timer               hostcfgd.service      
n/a                         n/a           Mon 2023-11-27 20:00:09 UTC 78ms ago      lldp.timer                   lldp.service             
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 274ms ago     lldp@0.timer                 lldp@0.service                
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 159ms ago     lldp@1.timer                 lldp@1.service                
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 337ms ago     mgmt-framework.timer         mgmt-framework.service        
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 216ms ago     pmon.timer                   pmon.service                
n/a                         n/a           Mon 2023-11-27 20:00:09 UTC 2ms ago       process-reboot-cause.timer   process-reboot-cause.service
n/a                         n/a           Mon 2023-11-27 20:00:09 UTC 60ms ago      snmp.timer                   snmp.service             
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 356ms ago     tacacs-config.timer          tacacs-config.service         
n/a                         n/a           Mon 2023-11-27 20:00:08 UTC 237ms ago     telemetry.timer              telemetry.service      
Mon 2023-11-27 20:10:00 UTC 9min left     Mon 2023-11-27 20:00:00 UTC 9s ago        logrotate.timer              logrotate.service           
Mon 2023-11-27 20:10:00 UTC 9min left     Mon 2023-11-27 20:00:00 UTC 9s ago        sysstat-collect.timer        sysstat-collect.service  
Mon 2023-11-27 22:36:29 UTC 2h 36min left Sun 2023-11-26 22:36:29 UTC 21h ago       systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Tue 2023-11-28 00:07:00 UTC 4h 6min left  Mon 2023-11-27 00:07:00 UTC 19h ago       sysstat-summary.timer        sysstat-summary.service       
Tue 2023-11-28 04:38:47 UTC 8h left       Mon 2023-11-27 11:08:02 UTC 8h ago        apt-daily.timer              apt-daily.service      
Tue 2023-11-28 06:19:57 UTC 10h left      Mon 2023-11-27 06:19:57 UTC 13h ago       apt-daily-upgrade.timer      apt-daily-upgrade.service   
Sun 2023-12-03 03:10:54 UTC 5 days left   Sun 2023-11-26 03:10:43 UTC 1 day 16h ago e2scrub_all.timer            e2scrub_all.service           
Mon 2023-12-04 00:00:00 UTC 6 days left   Mon 2023-11-27 00:00:00 UTC 20h ago       fstrim.timer                 fstrim.service                
n/a                         n/a           Wed 2023-11-22 22:22:47 UTC 4 days ago    rasdaemon.timer              rasdaemon.service      
                                                                                                                                          
19 timers listed.                                                                                                                            
Pass --all to see loaded but inactive timers, too.                                                                                      
root@sonic:/home/cisco# systemctl ^C                                                                                                           
root@sonic:/home/cisco# top                                                                                                             
top - 20:00:39 up 4 days, 21:39,  1 user,  load average: 0.15, 0.41, 0.74
Tasks: 198 total,   1 running, 193 sleeping,   0 stopped,   4 zombie
%Cpu(s):  0.0 us,  0.0 sy,  0.0 ni, 99.9 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :  31322.7 total,  27975.5 free,    808.9 used,   2538.3 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  30010.5 avail Mem

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                                  
    488 root      20   0 1862304  61600  30644 S   0.3   0.2  58:24.11 containerd
   1527 root      20   0  320792  98444  15416 S   0.3   0.3 224:45.22 appDemo
3945764 root      20   0   10912   3852   3288 R   0.3   0.0   0:00.02 top                                                                                                                                                                                      
      1 root      20   0  168152  13888   9036 S   0.0   0.0  10:29.59 systemd
```

Reply:
@anamehra what is your understanding of "OnUnitActiveSec". As I understand it, will keep firing the timer After the unit has become Active every `OnUnitActiveSec` seconds. Agree?

Reply:
I will check this on master. Do you know if 202305 has bookworm or plan to get it?
We will need this fix in 202305 and 202205.